client/src/reducers/socket.js

```diff
@@ -9,8 +9,8 @@ export default function SocketReducer(state = {
 			message: action.message
 		};
 	case 'SOCKET_NEW_TICKET': {
-		const newTicket = action.ticket;
-		newTicket.timestamp = new Date(newTicket.timestamp);
+		const newTicket = action.newTicket;
+		newTicket.timeFiled = new Date(newTicket.timeFiled);
```

Comment:
Hm so why do we need to make a Date object from newTicket.timeFiled? Is it not a Date object already?

Reply:
Please avoid corrupting the `action` object parameter (which this function doesn't own). Instead, make a new object like `{ ...newTicket, timeFiled: new Date(newTicket.timeFiled) }`.